data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/handlers/v1/beacon/GetStateFork.java

```diff
@@ -109,24 +109,23 @@ public void handle(@NotNull final Context ctx) throws Exception {
   public void handleRequest(RestApiRequest request) throws JsonProcessingException {
     final SafeFuture<Optional<ObjectAndMetaData<Fork>>> future =
         chainDataProvider.getFork(request.getPathParameter(PARAMETER_STATE_ID));
-    request.handleOptionalResult(future, this::handleResult, SC_NOT_FOUND);
-  }
-
-  private Optional<String> handleResult(Context ctx, final ObjectAndMetaData<Fork> response)
-      throws JsonProcessingException {
-    return Optional.of(
-        JsonUtil.serialize(
-            new StateForkData(response.isExecutionOptimisticForApi(), response.getData()),
-            RESPONSE_TYPE));
+    request.respondAsync(
+        future.thenApplyChecked(
```

Comment:
nit: I think just `thenApply` would be enough as I don't think there are any checked exceptions being thrown.

Reply:
fair point